include/parallel/parallel_implementation.h

```diff
@@ -2052,6 +2049,29 @@ inline status Communicator::probe (const unsigned int src_processor_id,
   return stat;
 }
 
+template<typename T>
+inline Status Communicator::packed_range_probe (const unsigned int src_processor_id,
+                                                const MessageTag & tag,
+                                                bool & flag) const
+{
+  START_LOG("packed_range_probe()", "Parallel");
+
+  Status stat((StandardType<typename Packing<T>::buffer_type>()));
+
+  int int_flag;
+
+  libmesh_call_mpi(MPI_Iprobe(src_processor_id,
+                              tag.value(),
+                              this->get(),
+                              &int_flag,
+                              stat.get()));
+
+  flag = int_flag;
+
+  STOP_LOG("packed_range_probe()", "Parallel");
+
+  return stat;
+}
 
```

Comment:
Could this be merged with the existing Communicator::probe() somehow?  The flag seems like something that ought to be encapsulated in a Status somehow and queryable after any probe.  The use of T to specify a datatype here seems unnecessarily premature; it ends up being used in Status::size() later, but at that point the data type is also specified and it would be possible to call size(data_type).